src/node_crypto.cc

```diff
@@ -4565,8 +4570,13 @@ void ECDH::SetPrivateKey(const FunctionCallbackInfo<Value>& args) {
   if (priv == nullptr)
     return env->ThrowError("Failed to convert Buffer to BN");
 
-  if (!EC_KEY_set_private_key(ecdh->key_, priv))
+
+  if (!EC_KEY_set_private_key(ecdh->key_, priv)) {
+    BN_free(priv);
     return env->ThrowError("Failed to convert BN to a private key");
+  }
+
+  BN_free(priv);
```

Comment:
The conditional runs something so that this can't be done before it?

Excuse my terrible pseudo-c++, but perhaps this could be better?:

```c++
var thingy = EC_KEY_set_private_key(ecdh->key_, priv);

BN_free(priv);
if (!thingy) {
   return env->ThrowError("Failed to convert BN to a private key");
}
```